### PR TITLE
feat: getTransactionHistory for portfolios

### PR DESCRIPTION
**Pull request by shuffledex**

### CODE REVIEW

File: src/api/entities/Portfolio/index.ts [hunk 1]
```diff
@@ -217,4 +221,83 @@ export class Portfolio extends Entity<UniqueIdentifiers> {
       return owner;
     }
   }
+
+  /**
+   * Retrieve a list of transactions where this portfolio was involved. Can be filtered using parameters
+   *
+   * @param filters.account - account involved in the transaction
```
Comment (monitz87) [inline src/api/entities/Portfolio/index.ts:228]: It isn't clear what this parameter represents. Which account? The sender or the receiver? Do we even want to expose this @VictorVicente ?

File: src/api/entities/Portfolio/index.ts [hunk 2]
```diff
@@ -217,4 +221,83 @@ export class Portfolio extends Entity<UniqueIdentifiers> {
...
+  ): Promise<ResultSet<HistoryData>> {
+    const {
+      context,
+      owner: { did },
+      _id,
+    } = this;
+
+    const { account, ticker, size, start } = filters;
+
+    /* eslint-disable @typescript-eslint/camelcase */
```
Comment (monitz87) [inline src/api/entities/Portfolio/index.ts:252]: Why is this here?

File: src/api/entities/Portfolio/index.ts [hunk 3]
```diff
@@ -217,4 +221,83 @@ export class Portfolio extends Entity<UniqueIdentifiers> {
...
+    } = result;
+
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const { items, totalCount: count } = settlementsResult!;
+
+    const data: HistoryData[] = [];
+    let next = null;
+
+    if (items) {
+      items.forEach(i => {
```
Comment (monitz87) [inline src/api/entities/Portfolio/index.ts:275]: `item` instead of `i`

File: src/api/entities/Portfolio/index.ts [hunk 4]
```diff
@@ -217,4 +221,83 @@ export class Portfolio extends Entity<UniqueIdentifiers> {
...
+        /* eslint-disable @typescript-eslint/no-non-null-assertion */
+        const { block_id, result: status, key, legs: settlementLegs } = i!;
```
Comment (monitz87) [inline src/api/entities/Portfolio/index.ts:277]: Rename `block_id` to `blockId`

File: src/api/entities/Portfolio/types.ts [hunk 5]
```diff
@@ -1,9 +1,23 @@
 import BigNumber from 'bignumber.js';
 
 import { SecurityToken } from '~/internal';
+import { SettlementDirectionEnum, SettlementResultEnum } from '~/middleware/types';
```
Comment (monitz87) [inline src/api/entities/Portfolio/types.ts:4]: Use type aliases for these two to rename them to remove the `Enum` at the end

File: src/api/entities/Portfolio/types.ts [hunk 6]
```diff
@@ -1,9 +1,23 @@
...
+export interface SettlementLeg {
+  token: SecurityToken;
+  amount: BigNumber;
+  direction: SettlementDirectionEnum;
+}
+
+export interface HistoryData {
+  blockNumber: BigNumber;
+  status: SettlementResultEnum;
+  account: string;
```
Comment (monitz87) [inline src/api/entities/Portfolio/types.ts:21]: This should be an `Account` object

File: src/api/entities/Portfolio/types.ts [hunk 7]
```diff
@@ -1,9 +1,23 @@
...
+  legs?: SettlementLeg[];
```
Comment (monitz87) [inline src/api/entities/Portfolio/types.ts:22]: Why is this optional?

File: src/api/entities/Portfolio/types.ts [hunk 8]
```diff
@@ -1,9 +1,23 @@
...
   locked: BigNumber;
 }
+
+export interface SettlementLeg {
+  token: SecurityToken;
+  amount: BigNumber;
+  direction: SettlementDirectionEnum;
+}
+
+export interface HistoryData {
```
Comment (monitz87) [inline src/api/entities/Portfolio/types.ts:18]: HistoricSettlement

**Review by monitz87 — CHANGES_REQUESTED** [review]

File: src/api/entities/Portfolio/index.ts (hunk 1 above)
Comment (VictorVicente) [reply src/api/entities/Portfolio/index.ts:228]: We should ask it to backend-team.

**Review by monitz87 — COMMENTED** [review]
Everything looks good, but we still need to hear from Product to see what we're going to do about the portfolios and account

File: src/api/entities/Proposal/types.ts [hunk 9]
```diff
@@ -2,7 +2,7 @@ import BigNumber from 'bignumber.js';
 import { TxTag } from 'polymesh-types/types';
 
 import { Account } from '~/internal';
-import { ProposalState } from '~/middleware/types';
+// import { ProposalState } from '~/middleware/types';
```
Comment (monitz87) [inline src/api/entities/Proposal/types.ts:5]: Use `ProposalStateEnum` instead of commenting out

File: src/api/entities/Portfolio/types.ts [hunk 10]
```diff
@@ -1,9 +1,25 @@
...
 }
+
+export interface SettlementLeg extends Leg {
+  direction: SettlementDirection;
+}
+
+export interface HistoricSettlement {
+  blockNumber: BigNumber;
+  status: SettlementResult;
+  accounts: Account[];
```
Comment (monitz87) [inline src/api/entities/Portfolio/types.ts:23]: Add a JSDOC comment above this property that says "array of accounts that participated by affirming the settlement"

File: src/middleware/queries.ts [hunk 11]
```diff
@@ -534,3 +535,49 @@ export function eventByAddedTrustedClaimIssuer(
     variables,
   };
 }
+
+/**
+ * @hidden
+ *
+ * Get settlements where a portfolio is envolved
```
Comment (monitz87) [inline src/middleware/queries.ts:542]:
```suggestion
 * Get Settlements where a Portfolio is involved
```

**Review by monitz87 — CHANGES_REQUESTED** [review]

File: src/api/entities/Portfolio/index.ts (hunk 1 above)
Comment (monitz87) [reply src/api/entities/Portfolio/index.ts:230]: @shuffledex change the description to "account involved in the settlement"

**Review by monitz87 — CHANGES_REQUESTED** [review]

**Review by monitz87 — APPROVED** [review]
